src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -68,7 +68,8 @@ DlgPrefSound::DlgPrefSound(QWidget* pParent,
             &DlgPrefSound::apiChanged);
 
     sampleRateComboBox->clear();
-    for (const auto& sampleRate : m_pSoundManager->getSampleRates()) {
+    const auto sampleRates = m_pSoundManager->getSampleRates();
+    for (const auto& sampleRate : sampleRates) {
```

Comment:
what warning does this fix? just out of curiosity because I don't see why one would be preferred over the other.

Reply:
In the new code the container is `const`.

Fixes: https://github.com/KDE/clazy/blob/master/docs/checks/README-range-loop-detach.md

Reply:
Ah right, I would've preferred `std::as_const(m_pSoundManager->getSampleRates())` instead though, without the extra variable. Otherwise it would make me think that `sampleRates` is used outside the loop as well. Any reason you decided against that?

Reply:
It's not possible though:

```cpp
src/preferences/dialog/dlgprefsound.cpp: In constructor ‘DlgPrefSound::DlgPrefSound(QWidget*, std::shared_ptr<SoundManager>, UserSettingsPointer)’:
src/preferences/dialog/dlgprefsound.cpp:71:48: error: use of deleted function ‘void std::as_const(const _Tp&&) [with _Tp = QList<mixxx::audio::SampleRate>]’
   71 |     for (const auto& sampleRate : std::as_const(m_pSoundManager->getSampleRates())) {
      |                                   ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt/QtCore/qglobal.h:47,
                 from /usr/include/qt/QtCore/qalgorithms.h:43,
                 from /usr/include/qt/QtCore/qdebug.h:44,
                 from /usr/include/qt/QtCore/QDebug:1,
                 from /home/jan/Projects/mixxx/src/audio/frame.h:3,
                 from /home/jan/Projects/mixxx/build/CMakeFiles/mixxx-lib.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/c++/13.2.1/utility:112:10: note: declared here
  112 |     void as_const(const _Tp&&) = delete;
      |          ^~~~~~~~
      ```

Reply:
Ah right, of course. thats probably why C++ now supports an init statement...
that has been shot down previously in code review (without good reason IMO) so I'm not suggesting to use it. 
```cpp
    for (const auto sampleRates = m_pSoundManager->getSampleRates(); const auto& sampleRate : sampleRates) {
```